luomu-libpcap/src/functions.rs

```diff
@@ -42,21 +42,16 @@ const PCAP_ERROR: i32 = libpcap::PCAP_ERROR;
 ///
 /// <https://www.tcpdump.org/manpages/pcap_create.3pcap.html>
 pub fn pcap_create(source: &str) -> Result<PcapT> {
-    let mut errbuf: Vec<u8> = vec![0; libpcap::PCAP_ERRBUF_SIZE as usize];
-    let interface = Some(source.to_string());
-    let source = CString::new(source)?;
+    let mut errbuf = Errbuf::new();
+    let interface = Some(source.into());
+    let src = CString::new(source)?;
 
-    let pcap_t =
-        unsafe { libpcap::pcap_create(source.as_ptr(), errbuf.as_mut_ptr() as *mut libc::c_char) };
-
-    trace!("pcap_create({:?}) => {:p}", source, pcap_t);
+    let pcap_t = unsafe { libpcap::pcap_create(src.as_ptr(), errbuf.as_mut_ptr()) };
 
+    trace!("pcap_create({}) => {:p}", source, pcap_t);
     if pcap_t.is_null() {
-        let cstr = unsafe { CStr::from_ptr(errbuf.as_ptr() as *const libc::c_char) };
-        let err = cstr.to_str()?.to_owned();
-        return Err(Error::PcapError(err));
+        return Err(Error::PcapError(errbuf.as_string()?));
```

Comment:
Could we add `as_error()` etc. to `Errbuf` as this pattern repeats. Might make the code a bit cleaner.

Reply:
Implemented in https://github.com/sensorfu/luomu-libpcap/pull/66/commits/1e21bc91ebad31cdfc50c0e0c05db3866e739f7e